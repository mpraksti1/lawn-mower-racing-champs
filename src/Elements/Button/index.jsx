import React from 'react';
import PropTypes from 'prop-types';
import StyledButton from './styles';

const Button = (props) => {
  const { children } = props;
  return (
    <StyledButton {...props}>{children}</StyledButton>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Button;
