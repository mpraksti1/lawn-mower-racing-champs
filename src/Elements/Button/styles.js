import styled, { css } from 'styled-components';

const StyledButton = styled.button`
  font-size: ${({ theme }) => theme.fontSizes.sm};
  font-family: ${({ theme }) => theme.fonts.baseFont};
  background-color: ${({ theme }) => theme.colors.darkGray};
  color: ${({ theme }) => theme.colors.white};
  padding: ${({ theme }) => theme.baseSizeUnit}px;
  cursor: pointer;
  display: block;
  outline: none;
  border: none;
  text-transform: uppercase;
  text-decoration: none;
  border-radius: 2px;
  margin: ${({
    spaceAround, theme, spaceAbove, spaceBelow,
  }) => {
    if (spaceAround) return `${theme.baseSizeUnit}px 0`;
    if (spaceAbove) return `${theme.baseSizeUnit}px 0 0 0`;
    if (spaceBelow) return `0 0 ${theme.baseSizeUnit}px 0`;
    return 0;
  }};

  /* For medium variant */
  ${({
    medium, theme,
  }) => medium && css`
    background: ${() => theme.fontSizes.md};
  `}

  /* For large variant */
  ${({
    large, theme,
  }) => large && css`
    font-size: ${() => theme.fontSizes.lg};
  `}

  /* For outline variant */
  ${({
    green, theme,
  }) => green && css`
    color: ${() => theme.colors.white};
    background: ${() => theme.colors.green};
    border: 1px solid ${() => theme.colors.green};
  `}

  /* For outline variant */
  ${({
    outline, theme,
  }) => outline && css`
    color: ${() => theme.colors.green};
    background: ${() => theme.colors.transparentBlack};
    border: 1px solid ${() => theme.colors.green};
  `}

  /* For inline variant */
  ${({ inline }) => inline && css`
    display: inline-block;
  `}

  /* For islink variant */
  ${({ islink }) => islink && css`
    display: inline;
    border: inherit;
    margin: inherit;
    padding: inherit;
    background: inherit;
    color: inherit;
    font: inherit;
    text-align: inherit;
    text-transform: inherit;
    outline: none;
    line-height: normal;
    -webkit-appearance: none;
  `}
`;

StyledButton.displayName = 'ButtonStyled';

export default StyledButton;
