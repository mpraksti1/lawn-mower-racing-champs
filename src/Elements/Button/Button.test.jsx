import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

describe('Button component', () => {
  it('renders without crashing', () => {
    shallow(<Button>Submit</Button>);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Button>Submit</Button>);

    expect(wrapper).toMatchSnapshot();
  });
});
