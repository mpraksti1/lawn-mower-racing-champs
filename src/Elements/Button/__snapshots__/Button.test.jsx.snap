// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<ButtonStyled>
  Submit
</ButtonStyled>
`;
